esi_leap/tests/objects/test_owner_change.py

```diff
@@ -0,0 +1,265 @@
+            'uuid': uuidutils.generate_uuid(),
+            'from_owner_id': 'owner1',
+            'to_owner_id': 'owner2',
+            'resource_type': 'test_node',
+            'resource_uuid': '12345',
+            'start_time': self.now,
+            'end_time': self.now + datetime.timedelta(days=100),
+            'fulfill_time': None,
+            'expire_time': None,
+            'status': statuses.ACTIVE,
+            'created_at': None,
+            'updated_at': None
+        }
+        self.config(lock_path=tempfile.mkdtemp(), group='oslo_concurrency')
+
+    @mock.patch('esi_leap.db.sqlalchemy.api.owner_change_get_by_uuid')
+    def test_get(self, mock_ocgbu):
+        oc_uuid = self.test_oc_data['uuid']
+        mock_ocgbu.return_value = self.test_oc_data
+
+        oc = owner_change.OwnerChange.get(oc_uuid, self.context)
+
+        mock_ocgbu.assert_called_once_with(oc_uuid)
+        self.assertEqual(self.context, oc._context)
+
+    @mock.patch('esi_leap.db.sqlalchemy.api.owner_change_get_all')
+    def test_get_all(self, mock_ocga):
+        mock_ocga.return_value = [self.test_oc_data]
+
+        ocs = owner_change.OwnerChange.get_all({}, self.context)
+
+        mock_ocga.assert_called_once_with({})
+        self.assertEqual(len(ocs), 1)
+        self.assertIsInstance(ocs[0], owner_change.OwnerChange)
+        self.assertEqual(self.context, ocs[0]._context)
+
+    @mock.patch('esi_leap.db.sqlalchemy.api.resource_verify_availability')
+    @mock.patch('esi_leap.db.sqlalchemy.api.owner_change_create')
+    def test_create(self, mock_occ, mock_rva):
+        oc = owner_change.OwnerChange(self.context, **self.test_oc_data)
+        mock_occ.return_value = self.test_oc_data
+
+        oc.create(self.context)
+
+        mock_rva.assert_called_once_with(oc.resource_type,
+                                         oc.resource_uuid,
+                                         oc.start_time,
+                                         oc.end_time,
+                                         is_owner_change=True)
+        mock_occ.assert_called_once_with(self.test_oc_data)
+
+    def test_create_invalid_time(self):
+        bad_oc_data = {
+            'id': 27,
+            'uuid': '534653c9-880d-4c2d-6d6d-11111111111',
+            'from_owner_id': 'owner1',
+            'to_owner_id': 'owner2',
+            'resource_type': 'test_node',
+            'resource_uuid': '12345',
+            'start_time': self.now + datetime.timedelta(days=100),
+            'end_time': self.now,
+            'status': statuses.CREATED,
+            'created_at': None,
+            'updated_at': None
+        }
+
+        oc = owner_change.OwnerChange(self.context, **bad_oc_data)
+
+        self.assertRaises(exception.InvalidTimeRange, oc.create)
+
+    @mock.patch('esi_leap.db.sqlalchemy.api.resource_verify_availability')
+    @mock.patch('esi_leap.db.sqlalchemy.api.owner_change_create')
+    def test_create_concurrent(self, mock_occ, mock_rva):
```

Comment:
I don't think this test proves create_owner_change is thread safe cus it's not a deterministic test.

Reply:
It seems deterministic to me... ? Perhaps it's worth looking into later, but I'm copying the pattern from test_lease.py and test_offer.py. I'm not sure there's a better way to test this.

Reply:
My thought is that assuming the function is prone to race conditions, the test may or may not detect this issue each time it's executed. But I'm not sure... It's may worth looking into later.

Reply:
I'll create an issue for looking at the concurrency tests!